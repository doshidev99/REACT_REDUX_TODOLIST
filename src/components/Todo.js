import React from 'react';
import PropTypes from 'prop-types';
import check from '../assets/images/check.png';
import checked from '../assets/images/checked.png';
import EditTodo from './EditTodo';

const Todo = ({
	toggleId,
	completed,
	text,
	removeId,
	onSave
}) => (
	<li>
		<img
			src={completed ? checked : check}
			onClick={toggleId}
			alt="todos"
		/>
		<EditTodo
			text={text}
			completed={completed}
			onSave={onSave}
		/>
		<button onClick={removeId}> Remove </button>
	</li>
);

Todo.propTypes = {
	completed: PropTypes.bool.isRequired,
	text: PropTypes.string.isRequired
};

export default Todo;
