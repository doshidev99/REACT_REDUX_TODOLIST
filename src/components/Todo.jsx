import React from 'react';
import PropTypes from 'prop-types';
import check from '../assets/images/check.png';
import checked from '../assets/images/checked.png';
import EditTodo from './EditTodo';

const Todo = ({ toggleId, completed, text, removeId, onSave }) => (
  <li>
    <img
      src={completed ? checked : check}
      onClick={toggleId}
      alt="todos"
      role="presentation"
    />
    <EditTodo text={text} completed={completed} onSave={onSave} />
    <button type="button" onClick={removeId}>
      Remove
    </button>
  </li>
);

Todo.propTypes = {
  completed: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
  toggleId: PropTypes.func.isRequired,
  removeId: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired,
};

export default Todo;
