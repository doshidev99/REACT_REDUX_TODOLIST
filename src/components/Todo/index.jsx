import React from 'react';
import PropTypes from 'prop-types';
import check from '../../assets/images/check.png';
import checked from '../../assets/images/checked.png';
import EditTodo from '../EditTodo';

const Todo = ({ toggleId, completed, text, removeId, onSave }) => (
  <li>
    <img
      src={completed ? checked : check}
      onClick={toggleId}
      alt="todos"
      role="presentation"
    />
    <EditTodo text={text} completed={completed} onSave={onSave} />
    <button type="button" onClick={removeId}>
      Remove
    </button>
  </li>
);

Todo.propTypes = {
  completed: PropTypes.bool,
  text: PropTypes.string,
  toggleId: PropTypes.func,
  removeId: PropTypes.func,
  onSave: PropTypes.func
};

Todo.defaultProps = {
  completed: false,
  text: '',
  toggleId: () => {},
  removeId: () => {},
  onSave: () => {}
};

export default Todo;
