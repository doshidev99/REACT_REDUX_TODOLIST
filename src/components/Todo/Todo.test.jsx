/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { shallow } from 'enzyme';
import Todo from './index';

describe('Todo', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Todo debug />);

    expect(component).toMatchSnapshot();
  });
});
