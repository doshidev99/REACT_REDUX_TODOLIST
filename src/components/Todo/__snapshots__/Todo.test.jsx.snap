// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should render correctly in "debug" mode 1`] = `
<li>
  <img
    alt="todos"
    onClick={[Function]}
    role="presentation"
    src="check.png"
  />
  <EditTodo
    completed={false}
    onSave={[Function]}
    text=""
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Remove
  </button>
</li>
`;
